tracer/src/Datadog.Trace/Debugger/Expressions/ProbeExpressionParser.Dump.cs

```diff
@@ -0,0 +1,289 @@
+        }
+
+        var stringConcat = GetMethodByReflection(typeof(string), nameof(string.Concat), new[] { typeof(object[]) });
+
+        if (IsMicrosoftException(expression.Type))
+        {
+            var typeNameExpression = Expression.Constant(expression.Type.FullName, typeof(string));
+            var ifNull = Expression.Equal(expression, Expression.Constant(null));
+            var exceptionAsString = Expression.Call(
+               stringConcat,
+               Expression.NewArrayInit(
+                   typeof(string),
+                   typeNameExpression,
+                   Expression.Constant(", "),
+                   Expression.Property(expression, nameof(Exception.Message)),
+                   Expression.Constant(", "),
+                   Expression.Property(expression, nameof(Exception.StackTrace))));
+
+            return Expression.Condition(ifNull, typeNameExpression, exceptionAsString);
+        }
+
+        return IsSafeCollection(expression.Type) ?
+                   DumpCollectionExpression(expression, scopeMembers) :
+                   DumpFieldsExpression(expression, scopeMembers);
+    }
+
+    private Expression DumpCollectionExpression(Expression collection, List<ParameterExpression> scopeMembers)
+    {
+        var loopItemType = collection.Type.IsGenericType ? collection.Type.GetGenericArguments()[0] : typeof(object);
+        var enumerableType = typeof(IEnumerable<>).MakeGenericType(loopItemType);
+        var enumeratorType = typeof(IEnumerator<>).MakeGenericType(loopItemType);
+        var stringBuilderAppend = GetMethodByReflection(typeof(StringBuilder), nameof(StringBuilder.Append), new[] { typeof(string) });
+        var getEnumeratorCall = Expression.Call(collection, GetMethodByReflection(enumerableType, nameof(IEnumerable.GetEnumerator), Type.EmptyTypes));
+
+        var expressions = new List<Expression>();
+        var loopItem = Expression.Variable(loopItemType, "loopItem");
+        scopeMembers.Add(loopItem);
+
+        var enumeratorVar = Expression.Variable(enumeratorType, "enumerator");
+        scopeMembers.Add(enumeratorVar);
+        expressions.Add(enumeratorVar);
+        var moveNextCall = Expression.Call(enumeratorVar, GetMethodByReflection(typeof(IEnumerator), nameof(IEnumerator.MoveNext), Type.EmptyTypes));
+
+        expressions.Add(Expression.Assign(enumeratorVar, getEnumeratorCall));
+
+        var index = Expression.Variable(typeof(int), "index");
+        scopeMembers.Add(index);
+        expressions.Add(Expression.Assign(index, Expression.Constant(0)));
+
+        var result = Expression.Variable(typeof(StringBuilder), "itemValues");
+        scopeMembers.Add(result);
+        expressions.Add(Expression.Assign(result, Expression.New(typeof(StringBuilder).GetConstructor(Type.EmptyTypes))));
+
+        expressions.Add(Expression.Call(result, stringBuilderAppend, Expression.Constant("[")));
+
+        var dumpObjectCallExpression = Expression.Call(
```

Comment:
Could we get bitten by the version conflict bug in this invocation?

Reply:
Discussed offline  - according to Dudi we expect that each compiled expression will reference the Datadog.Trace that created it, so we should be good.